### ISSUE TITLE
Linear Operator Class

### ISSUE THREAD

**Issue by JonathanWenger** [probabilistic-numerics/probnum#1]
Use existing or implement a linear operator class similar to `scipy.sparse.linalg.LinearOperator` which implements methods like `.transpose()` and `todense()` even for sums and concatenations of operators.

**Comment by JonathanWenger** [comment]
It could be more efficient to _not_ perform operations with ``MatrixLinearOperator`` lazily.

**Comment by JonathanWenger** [comment]
We need our own implementation (which subclasses either the ``scipy`` or the ``pylops`` implementation) to allow binary arithmetic operations with ``RandomVariables`` such as:
```
import numpy as np
from probnum.probability import RandomVariable, Normal
from probnum.linalg import aslinearoperator

# Linear operators applied to random variables
A = aslinearoperator(np.array([[1, 2], [3, 4]]))
X = RandomVariable(distribution=Normal(mean=np.array([1, 2]), cov=np.array([[2, 0], [0, 5]])))
Y = A @ X
```

**Comment by JonathanWenger** [comment]
Skeleton making ``LinearOperator.dot`` suitable for ``RandomVariables`` implemented by 6892e8027782680ad99fda768855363e6c28b612.

**Comment by JonathanWenger** [comment]
``.transpose()`` and ``.todense()`` for binary arithmetic operations implemented by f99f573b6febd9569b6e940a9b73a3dde458ffc7.

**Comment by JonathanWenger** [comment]
Custom linear operator creation from a function ``matvec`` implemented by 9a68462.
